modules/porous_flow/include/materials/PorousFlowFluidStateSingleComponent.h

```diff
@@ -0,0 +1,122 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef POROUSFLOWFLUIDSTATESINGLECOMPONENT_H
+#define POROUSFLOWFLUIDSTATESINGLECOMPONENT_H
+
+#include "PorousFlowVariableBase.h"
+#include "PorousFlowFluidStateSingleComponentBase.h"
+
+class PorousFlowFluidStateSingleComponent;
+class PorousFlowCapillaryPressure;
+
+template <>
+InputParameters validParams<PorousFlowFluidStateSingleComponent>();
+
+/**
+ * Fluid state class using a persistent set of primary variables for
+ * the mutliphase, single component case.
+ *
+ * Primary variables are: liquid pressure and enthalpy.
+ *
+ * The PorousFlow kernels expect saturation and mass fractions (as well as pressure
+ * and temperature), so these must be calculated from the primary variables once the
+ * state of the system is determined.
+ */
+class PorousFlowFluidStateSingleComponent : public PorousFlowVariableBase
+{
+public:
+  PorousFlowFluidStateSingleComponent(const InputParameters & parameters);
+
+protected:
+  virtual void initQpStatefulProperties() override;
+  virtual void computeQpProperties() override;
+
+  /// Size material property vectors and initialise with zeros
+  void setMaterialVectorSize() const;
+
+  /**
+   * Calculates all required thermophysical properties and derivatives for each phase
+   * and fluid component. Must override in all derived classes.
```

Comment:
You say this must be overridden, but you provide an implementation. That can confuse your users...

Reply:
Good catch. This was a copy and paste as well, but the other one should be fixed as well. This method doesn't need to be overridden in most cases